google/cloud/storage/async_object_responses.h

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_ASYNC_OBJECT_RESPONSES_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_ASYNC_OBJECT_RESPONSES_H
+
+#include "google/cloud/storage/object_metadata.h"
+#include "google/cloud/storage/version.h"
+#include "absl/types/optional.h"
+#include <cstdint>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace google {
+namespace cloud {
+namespace storage_experimental {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+/// Represents the response from reading a subset of an object.
+struct ReadObjectRangeResponse {
```

Comment:
Does this have any relation to `internal::ReadObjectRangeRequest`? If no, are you concerned that a future developer will trip on these names?

Reply:
Hmmm... I added an `Async` prefix to avoid that.